ports/geckolib/glue.rs

```diff
@@ -3620,10 +3619,11 @@ pub extern "C" fn Servo_ReparentStyle(
     let element = element.map(GeckoElement);
 
     let mut cascade_flags = CascadeFlags::empty();
-    if style_to_reparent.is_anon_box() {
-        cascade_flags.insert(CascadeFlags::SKIP_ROOT_AND_ITEM_BASED_DISPLAY_FIXUP);
-    }
     if let Some(element) = element {
+        // NOTE(emilio): This relies on element.is_none() == pseudo.is_none(),
+        // which the caller guarantees, fortunately. But this doesn't handle the
+        // IS_ROOT_ELEMENT flag correctly! I think it's not possible to wrap a
+        // root element in a first-line frame, so this may be fine.
```

Comment:
It is unclear to me how this is related to first-line frame... Could you elaborate it?

Reply:
Reparenting in Servo only happens to restyle `::first-line`.

Reply:
I mentioned why this is related to ::first-line.